src/pages/studyView/resources/FilesAndLinks.tsx

```diff
@@ -112,13 +125,52 @@ async function fetchFilesLinksData(
         0
     );
 
-    const resourcesForPatients = await getResourceDataOfPatients(
-        studyClinicalDataResponse.data
+    const selectedStudyIds = [
+        ...new Set(selectedSamples.map(item => item.studyId)),
+    ];
+    const selectedSampleIds = selectedSamples.map(item => item.sampleId);
+    const selectedPatientIds = [
+        ...new Set(selectedSamples.map(item => item.patientId)),
+    ];
+
+    const resourcesForEntireStudy = await getResourceDataOfEntireStudy(
+        selectedStudyIds
     );
-    const resourcesForPatientsAndSamples: { [key: string]: ResourceData[] } = {
-        ...sampleIdResourceData,
-        ...resourcesForPatients,
-    };
+
+    const resourcesForPatientsAndSamples = resourcesForEntireStudy
+        // Filter the resources to consist of only studyView selected samples
+        // Also keep patient level resources (e.g. Those don't have a sampleId)
+        .filter(
```

Comment:
one more thing here:  this is operation is R*samples+R*patients.  could easily get to billion.    easily optimized by putting selectedSamples in a map so you can use IN operator.

Reply:
I have replaced this with Maps instead of arrays to remove the linear lookup. I don't think you can use the IN operator with Maps so we're opting for the .has('key') expression instead.